Add TextInfo casing tests

ellismg opened the pull request:
An initial set of TextInfo casing tests. I looked at the scinerios we
have coverage for in ToF, but wrote the actual tests from scratch in
idomatic XUnit style.

In addition, I have added some specific tests for the Turkish I casing,
since the characters have special casing behavior in Turkish and some
other locales as well as test cases to ensure we have the simple casing
in places where the Unicode standard normally defines special cases
which are either context sensitive or would change the number of
codepoints in a cased string.

ellismg commented:
/cc @stephentoub.  These are still disabled on Unix for now, since I have not landed the ICU based casing stuff yet.

stephentoub commented on src/System.Globalization/tests/TextInfo.cs:
```diff
@@ -0,0 +1,143 @@
+        Assert.Equal('1', ti.ToUpper('1'));
+        Assert.Equal("123", ti.ToUpper("123"));
+
+        Assert.Equal("EMBEDED\0NULL\0BYTE\0", ti.ToUpper("embeded\0NuLL\0Byte\0"));
```
Extreme nit: embedded

ellismg replied:
Fixed.

stephentoub commented on src/System.Globalization/tests/TextInfo.cs:
```diff
@@ -0,0 +1,143 @@
+        Assert.Equal('\u00D3', ti.ToUpper('\u00F3'));
+        Assert.Equal("\u00D3", ti.ToUpper("\u00F3"));
+
+        Assert.Equal('\u2603', ti.ToUpper('\u2603'));
```
For these specific unicode character tests, is it worth adding a comment indicating what it is / why it's chosen?

ellismg replied:
Can do (the first is a character that actually cases (an o with an accent mark), the second is a character that does not (a snowman)).

ellismg replied:
I also added a test for a character outside the BMP that does have a casepair.

stephentoub commented on src/System.Globalization/tests/System.Globalization.Tests.csproj:
```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Could you add a .sln with this as well?

ellismg replied:
Can do.

stephentoub commented:
A few minor comments, but otherwise LGTM. Thanks!

ellismg commented:
Thanks Steve, I addressed all your feedback, hopefully fixed the issue that was causing the change to fail in CI (which I didn't hit locally because I am on Windows 10) and squashed.

stephentoub commented:
Thanks, Matt.  Assuming it passes, LGTM.
